### PR TITLE
Headtail Storage

### PR DESCRIPTION
**Pull request by Geevies**
* Skrell can now store a single small item in their headtails via an ability in the top left corner of their screen.

Loosely ported from https://github.com/ParadiseSS13/Paradise/pull/7377. Stunning doesn't make them drop their item here.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -399,6 +399,7 @@
 		dat += "<BR><A href='?src=\ref[src];item=tie'>Remove accessory</A>"
 	dat += "<BR><A href='?src=\ref[src];item=splints'>Remove splints</A>"
 	dat += "<BR><A href='?src=\ref[src];item=pockets'>Empty pockets</A>"
+	dat += species.handle_strip("\ref[src]")
```
Comment (Arrow768) [inline code/modules/mob/living/carbon/human/human.dm:402]: I am not a fan of that implementation.
Here you use a species proc to determine the text that should be displayed.

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -585,6 +586,16 @@
...
+	if(href_list["headtail"])
+		usr.visible_message(SPAN_WARNING("\The [usr] is trying to remove something from \the [src]'s headtails!"))
+		if(do_after(usr, HUMAN_STRIP_DELAY, act_target = src))
+			var/obj/item/storage/internal/skrell/S = locate() in src
+			var/obj/item/I = locate() in S
+			if(I)
+				S.remove_from_storage(I, get_turf(src))
+			else
+				to_chat(usr, SPAN_WARNING("\The [src] had nothing in their headtail storage."))
+
```
Comment (Arrow768) [inline code/modules/mob/living/carbon/human/human.dm:598]: And here you hard-code the button that is being pressed.

This should not be hardcoded, but rather handled on a species level.
I.e. with a additional parameter species_specific in the clicked link which then causes a "species handler" to handle that interaction.

**Review by Arrow768 — DISMISSED** [review]

**Review by Alberyk — DISMISSED** [review]
Need lore writer approval.

File: code/modules/organs/subtypes/skrell.dm [hunk 3]
```diff
@@ -16,3 +16,20 @@
 
 /obj/item/organ/internal/brain/skrell
 	icon_state = "brain_skrell"
+
+/obj/item/storage/internal/skrell
```
Comment (mikomyazaki) [inline code/modules/organs/subtypes/skrell.dm:20]: Currently this implementation doesn't seem to drop the item when the head (and presumably the tails attached) are destroyed.

I think a better way would be to make the storage be inside their head organ, and have the head organ's destroy handle dropping the item within.

Also, this file `code/modules/organs/subtyles/skrell.dm` is for organs -- This is not an organ.

**Review by mikomyazaki — CHANGES_REQUESTED** [review]

**Comment by Geevies** [comment]
gonna reopen this when a skrell maintainer is chosen and they confirm this

**Comment by Geevies** [comment]
image

!review

**Review by NonQueueingMatt — APPROVED** [review]

**Review by skull132 — APPROVED** [review]

**Pull request merged**